src/app/redux/TransactionReducer.js

```diff
@@ -0,0 +1,123 @@
+import {fromJS, Map} from 'immutable';
+
+const defaultState = fromJS({
+    operations: [],
+    status: { key: '', error: false, busy: false, },
+    errors: null
+});
+
+export default function reducer(state = defaultState, action) {
+    const payload = action.payload;
+
+    if (action.type === 'transaction/CONFIRM_OPERATION') {
+        const operation = fromJS(payload.operation);
+        const confirm = payload.confirm;
+        const warning = payload.warning;
+        return state.merge({
+            show_confirm_modal: true,
+            confirmBroadcastOperation: operation,
+            confirmErrorCallback: payload.errorCallback,
+            confirm,
+            warning
+        });
+    }
+
+    if (action.type === 'transaction/HIDE_CONFIRM') {
+        return state.merge({show_confirm_modal: false, confirmBroadcastOperation: undefined, confirm: undefined});
+    }
+
+    if (action.type === 'transaction/BROADCAST_OPERATION') {
```

Comment:
These aren't necessary (but are kind of nice if you want to be explicit about the actions being thrown, even though you don't handle all of em)

Reply:
Yes, unfortunately we don't have any meaningful tests coverage, this is going to be changed in soon - now we have more resources now compared to a time when we were starting. I can understand you concerns about quite a lot of different changes going into this single branch. We have thoroughly tested this branch so it should be good to go for now. In the future we are planning to put more efforts into automated testing and any help on this is very welcome!

Reply:
Ahh I totally understand. Under a time crunch tests are often the first things to get cut!
If I can find an hour or two this weekend I'll cut a branch and implement Jest :)